ckanext/datastore/backend/postgres.py

```diff
@@ -330,48 +330,100 @@ def _get_fields(connection: Any, resource_id: str):
 
 # (canada fork only): foreign keys
 # TODO: upstream contrib!!
-def _get_foreign_constraints(connection, resource_id, return_constraint_names=False,
-                             column_name=None):
-    u'''
-    return a list of foreign constraint names.
+def _get_foreign_constraints(connection, resource_id, return_constraint_names=False):
     '''
-    # p.conrelid: The table this constraint is on
-    # p.conindid: The index supporting this constraint
-    # p.confrelid: The referenced table
-    # p.confupdtype: Foreign key update action code (#TODO: support `cascade`??)
-    # p.confdeltype: Foreign key deletion action code (#TODO: support `cascade`??)
-    # p.conkey: List of the constrained columns (int2, pg_attribute.attnun {The number of the column})
-    # p.confkey: List of the referenced columns (int2, pg_attribute.attnun {The number of the column})
-    #FIXME: this is just super broken.
-    foreign_constraints = {} if not return_constraint_names else []
-    foreign_constraints_sql = sa.text(u'''
-        SELECT
-        p.conname as constraint_name,
-        c1.relname as foreign_table,
-        co1.column_name as foreign_column,
-        co2.column_name as primary_column
-        FROM pg_constraint p
-        LEFT JOIN pg_class c1 ON c1.oid = p.confrelid
-        LEFT JOIN pg_class c2 ON c2.oid = p.conrelid
-        JOIN information_schema.columns co1 ON
-            (SELECT relname FROM pg_class WHERE oid = p.confrelid) = co1.table_name
-            AND co1.ordinal_position = ANY(p.confkey)
-        JOIN information_schema.columns co2 ON
-            (SELECT relname FROM pg_class WHERE oid = p.conrelid) = co2.table_name
-            AND co2.ordinal_position = ANY(p.conkey)
-        WHERE p.contype = 'f'
-            AND c2.relname = '{0}'
-            {1}
-        ORDER BY p.oid;
-    '''.format(resource_id,
-               "AND co1.column_name = '{}'".format(column_name) if column_name else ''))
-    foreign_constraints_results = connection.execute(foreign_constraints_sql)
-    for result in foreign_constraints_results.fetchall():
-        if return_constraint_names:
+    Return a list of foreign constraint names or the info for the constraints.
+    '''
+    foreign_constraints = []
+    if return_constraint_names:
+        foreign_constraints_sql = sa.text('''
+            SELECT
+            p.conname as constraint_name
+            FROM pg_constraint p
+            LEFT JOIN pg_class c1 ON c1.oid = p.confrelid
+            LEFT JOIN pg_class c2 ON c2.oid = p.conrelid
+            WHERE p.contype = 'f'
+                AND c2.relname = '{0}'
+            ORDER BY p.oid;
+        '''.format(resource_id))
+        foreign_constraints_results = connection.execute(foreign_constraints_sql)
+        for result in foreign_constraints_results.fetchall():
             foreign_constraints.append(result.constraint_name)
```

Comment:
loops are slow, should be fine to
```suggestion
        foreign_constraints += foreign_constraints_results.fetchall()
```